### ISSUE TITLE
Remove internal TSMP version number (-v)

### ISSUE THREAD

**Issue by s-poll** [HPSCTerrSys/TSMP#147]
## Remove internal version number

I would suggest to remove the internal version number, which is indicated by `-v` option in `build_tsmp` and `setup_tsmp`. The internal version number does not have a particular purpose and only partly serves to distinguish the component models used for building etc. Since TSMP tag >=v1.3.x, the internal version number is not bi-unique connected to component version number and thus, the interval version number may lead to confusion for the users. 

I would suggest to supplement the controlling of the component version by an extension of the combination option (`-c`) with the component model version, e.g. `-c clm3_5-cos5_1-pfl3_9`. Components could be separated by `-` as before, components are added after the component name. The fine-grading of the component model version would increase in visibility and it would be still possible to manage within TSMP the building/setting up procedure. The availability could be managed by the combinations string to ensure that the model configuration is available on the specific machine.

**Comment by jjokella** [comment]
I agree with this proposal completely.

However, we should be cautious about removing the `-v`  option completely, since the option will appear in many scripts.

Regarding backward compatibility, I think it will be hard to implement a backward compatible way of handling these options that is more lightweight than the current situation...

Nevertheless, it might be worth complicating things a little bit first in order to  solve the uniqueness/confusion issues you mentioned.

**Comment by s-poll** [comment]
I agree that removing the `-v` option should be done with caution. However, the option is only used twice (once in the `build_tsmp.ksh` and once in `setup_tsmp.ksh`) to translate into `mList`parameter. My idea is to keep the `mList` parameter but controlling the parameter by the use of the `-c` option. 

Regarding backward compatibility. This is indeed complex to implement. I would suggest to implement an error-message in the build/setup_tsmp, if someone uses the `-v` option including instructions how to manage component model versions.

**Comment by chartick** [comment]
The internal version number is removed in release_candidate. Now it is required to provide a combination with -c in the form of clm3-cos5-pfl-pdaf. At the moment, I am testing if this works for all cases.

**Comment by niklaswr** [comment]
Isn't it counterintuitive or even confusing to have a combination flag in the form of `clm3-cos5-pfl-pdaf`?

I particularly disagree with `pdaf`. PDAF is more of an *interface* similar to OASIS, and OASIS is not part of the combination flag as well, is it?  
How about an extra flag to control PDAF: 
`-c clm3-cos5-pfl --pdaf TRUE`?

**Comment by chartick** [comment]
The idea was to integrate the internal version number into the combination. In the old workflow, PDAF was an addendum to selected internal versions, therefore I added it to the combination. We could instead add another letter, but personally I think there are already enough. If PDAF is not in the combination it would also be possible to add it to unsupported combinations, for example something with ICON, although this is a minor problem.

I would have done the same for OASIS, but we removed the OASIS (without MCT) support, so it is included in any combination and therefore unnecessary to add.

**Comment by DCaviedesV** [comment]
@niklaswr, although I agree that PDAF is more of an interface than a component, I think the main issue is that OASIS is kind of mandatory, whereas PDAF is not. Of course, one could argue that the OASIS version could be specified, but currently, as @chartick this is not a problem. This is a bit of a model/code taxonomy question, I think. From the point of view of the user, what I think mostly matters is being able to easily include PDAF or not, and, IMHO, the exact taxon for each component/interface is not particularly in the users mind.
I don´t have anything in particular against the idea of setting an extra flag such as `--pdaf`, but indeed there are already already way to many short options to parse... which makes adding new ones annoying, as you end up choosing an unrelated letter. I would discourage this.
As far as I remember long options are not fully supported with vanilla `getopts`, but I may be wrong. If they are easily supported, then the `--pdaf` would be something to consider, but if it requires some overhaul of the `getopts` parsing, then I would suggest not to this at this time, but later on with the upcoming `cmake` infrastructure.

**Comment by niklaswr** [comment]
In principle I still think it is not strict to include `pdaf` but not `oasis` (or none of both) in the combination string, but in general I agree with your argumentation. 

Long options are possible with the currently implemented `getopts`, but only in combination with a short option. So yes, considering this with the new `cmake` version might be reasonable.

**Comment by chartick** [comment]
The internal version number is not needed anymore in the master, so that we can close this for now.
